packages/commonwealth/vite.config.ts

```diff
@@ -0,0 +1,124 @@
+      react(),
+      createHtmlPlugin({
+        entry: `./${command === 'serve' ? 'client/' : ''}scripts/index.tsx`,
+        template: './client/index.html',
+      }),
+      tsconfigPaths(),
+    ],
+    publicDir: 'static',
+    server: {
+      port: 8080,
+      host: 'localhost',
+      proxy: {
+        '/api': {
+          target: env.BACKEND_PROXY_URL || 'http://localhost:3000',
+          changeOrigin: true,
+          secure: false,
+        },
+      },
+    },
+    resolve: {
+      alias: [
+        {
+          // matches only non-relative paths that end with .scss
+          find: /^([^.].*)\.scss$/,
+          replacement: path.resolve(
+            projectRootDir,
+            'client',
+            'styles',
+            '$1.scss',
+          ),
+        },
+        createClientResolver('hooks'),
+        createClientResolver('navigation'),
+        createClientResolver('state'),
+        createClientResolver('views'),
+        createClientResolver('controllers'),
+        createClientResolver('models'),
+        createClientResolver('helpers'),
+      ],
+      extensions: ['.ts', '.tsx', '.js', '.jsx'],
+    },
+    define: {
+      global: {},
+      'process.env.NODE_ENV': JSON.stringify('development'),
+      'process.env.SERVER_URL': JSON.stringify(env.SERVER_URL),
+      'process.env.KNOCK_PUBLIC_API_KEY': JSON.stringify(
+        env.KNOCK_PUBLIC_API_KEY,
+      ),
+      'process.env.KNOCK_IN_APP_FEED_ID': JSON.stringify(
+        env.KNOCK_IN_APP_FEED_ID,
+      ),
+      'process.env.MIXPANEL_DEV_TOKEN':
+        JSON.stringify(env.MIXPANEL_DEV_TOKEN) ||
+        JSON.stringify('312b6c5fadb9a88d98dc1fb38de5d900'),
+      'process.env.MAGIC_PUBLISHABLE_KEY':
+        JSON.stringify(env.MAGIC_PUBLISHABLE_KEY) ||
+        JSON.stringify('pk_live_EF89AABAFB87D6F4'),
+      'process.env.DISCORD_CLIENT_ID':
+        JSON.stringify(env.DISCORD_CLIENT_ID) ||
+        JSON.stringify('1034502265664454776'),
+      'process.env.DISCORD_UI_URL':
+        JSON.stringify(env.DISCORD_UI_URL) ||
+        JSON.stringify('http://localhost:3000'),
+      'process.env.COSMOS_GOV_V1': JSON.stringify(env.COSMOS_GOV_V1),
+      'process.env.COSMOS_REGISTRY_API': JSON.stringify(
+        env.COSMOS_REGISTRY_API,
+      ),
+      'process.env.FLAG_COMMUNITY_HOMEPAGE': JSON.stringify(
+        env.FLAG_COMMUNITY_HOMEPAGE,
+      ),
+      'process.env.FLAG_PROPOSAL_TEMPLATES': JSON.stringify(
+        env.FLAG_PROPOSAL_TEMPLATES,
+      ),
+      'process.env.FLAG_CONTEST': JSON.stringify(env.FLAG_CONTEST),
+      'process.env.ETH_RPC': JSON.stringify(env.ETH_RPC),
+      'process.env.FLAG_COMMUNITY_STAKE': JSON.stringify(
+        env.FLAG_COMMUNITY_STAKE,
+      ),
+      'process.env.FLAG_USER_ONBOARDING_ENABLED': JSON.stringify(
+        env.FLAG_USER_ONBOARDING_ENABLED,
+      ),
+      'process.env.FLAG_ALLOWLIST': JSON.stringify(env.FLAG_ALLOWLIST),
+      'process.env.IS_PRODUCTION': JSON.stringify(env.IS_PRODUCTION),
```

Comment:
Off-topic: why aren't we using `IS_PRODUCTION` and not `NODE_ENV`?

Reply:
Well NODE_ENV is a little misleading since technically the client will not be using node (It just runs regular javascipt interpreted by the browser). But anyways this is the same variable in webpack, so I am just copying it over here.